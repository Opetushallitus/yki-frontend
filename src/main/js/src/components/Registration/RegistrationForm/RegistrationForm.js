import { FinnishSSN } from 'finnish-ssn';
import { ErrorMessage, Field, Form, Formik } from 'formik';
import { parsePhoneNumberFromString } from 'libphonenumber-js';
import moment from 'moment';
import PropTypes from 'prop-types';
import React from 'react';
import { withTranslation } from 'react-i18next';
import * as Yup from 'yup';

import { DATE_FORMAT, ISO_DATE_FORMAT_SHORT } from '../../../common/Constants';
import ScrollToError from '../../../ScrollToFormTop';
import { useMobileView } from '../../../util/customHooks';
import { checkBirthDate, containsSpecialCharacters } from '../../../util/util';
import FormikInputField from '../../FormikInputField/FormikInputField';
import { useTranslation } from 'react-i18next';

import PhoneNumberInput from '../../PhoneNumberInput/PhoneNumberInput';
import Button from '../../UI/Button/Button';
import Checkbox from '../../UI/Checkbox/Checkbox';
import RadioButton from '../../UI/RadioButton/RadioButton';
import ZipAndPostOffice from '../../ZipAndPostOffice/ZipAndPostOffice';
import RegistrationError from '../RegistrationError/RegistrationError';
import GenderSelect from './GenderSelect/GenderSelect';
import NationalitySelect from './NationalitySelect/NationalitySelect';
import classes from './RegistrationForm.module.css';

const ExternalLink = ({ label, url }) => {
  const { t } = useTranslation();

  return (
    <a
      href={url}
      target="_blank"
      rel="noopener noreferrer"
      aria-label={t('common.newTab')}
    >
      {label}
      <img
        className={classes.ExternalLinkIcon}
        src={require('../../../assets/svg/external-link.svg')}
        alt={t('common.newTab')}
      />
    </a>
  );
};

export const registrationForm = props => {
  const mandatoryErrorMsg = props.t('error.mandatory');
  const maxErrorMsg = props.t('error.max');
  const mobileOrTablet = useMobileView(true, true);

  function validatePhoneNumber(value) {
    if (value) {
      const phoneNumber = parsePhoneNumberFromString(value);
      return phoneNumber && phoneNumber.isValid();
    } else {
      return false;
    }
  }

  function validateSsn(value) {
    return !value || FinnishSSN.validate(value);
  }

  function validateBirthDate(value) {
    if (props.initData.user.ssn) {
      return true;
    }
    const validation = checkBirthDate(value);
    if (validation.error) {
      return this.createError({ message: validation.error });
    }
    return true;
  }

  function sameEmail(confirmEmail) {
    return confirmEmail === this.parent.email;
  }

  const validationSchema = Yup.object().shape({
    firstName: Yup.string()
      .required(mandatoryErrorMsg)
      .max(128, maxErrorMsg)
      .test(
        'no-special-characters',
        props.t('error.specialCharacters'),
        value => !containsSpecialCharacters(value),
      ),
    lastName: Yup.string()
      .required(mandatoryErrorMsg)
      .max(128, maxErrorMsg)
      .test(
        'no-special-characters',
        props.t('error.specialCharacters'),
        value => !containsSpecialCharacters(value),
      ),
    streetAddress: Yup.string()
      .required(mandatoryErrorMsg)
      .max(128, maxErrorMsg)
      .test(
        'no-special-characters',
        props.t('error.specialCharacters'),
        value => !containsSpecialCharacters(value),
      ),
    zip: Yup.string()
      .required(mandatoryErrorMsg)
      .max(16, maxErrorMsg),
    postOffice: Yup.string()
      .required(mandatoryErrorMsg)
      .max(64, maxErrorMsg)
      .test(
        'no-special-characters',
        props.t('error.specialCharacters'),
        value => !containsSpecialCharacters(value),
      ),
    phoneNumber: Yup.string()
      .required(mandatoryErrorMsg)
      .test(
        'invalid-phone-number',
        props.t('error.phoneNumber'),
        validatePhoneNumber,
      ),
    email: Yup.string()
      .email(props.t('error.email'))
      .required(mandatoryErrorMsg)
      .max(64, maxErrorMsg),
    nationality: Yup.string()
      .required(mandatoryErrorMsg)
      .test('gender-select', mandatoryErrorMsg, value => {
        return value && value !== 'placeholder';
      }),
    gender: Yup.string(),
    ssn: Yup.string().test(
      'invalid-ssn',
      props.t('error.ssn.invalid'),
      validateSsn,
    ),
    confirmEmail: Yup.string().test(
      'same-email',
      props.t('error.confirmEmail'),
      sameEmail,
    ),
    birthdate: Yup.string().test(
      'invalid-birthdate',
      props.t('error.birthdate'),
      validateBirthDate,
    ),
    examLang: Yup.string().required(mandatoryErrorMsg),
    certificateLang: Yup.string().required(mandatoryErrorMsg),
    personalDataConsent: Yup.boolean()
      .required(mandatoryErrorMsg)
      .oneOf([true], mandatoryErrorMsg),
    termsOfUseConsent: Yup.boolean()
      .required(mandatoryErrorMsg)
      .oneOf([true], mandatoryErrorMsg),
  });

  const RadioButtonComponent = ({
    field: { name, value, onChange },
    id,
    checkedValue,
    label,
  }) => {
    return (
      <RadioButton
        name={name}
        id={id}
        checkedValue={checkedValue}
        value={value}
        onChange={onChange}
        label={label}
      />
    );
  };

  const RadioButtonGroup = ({
    value,
    id,
    label,
    className,
    children,
    error,
  }) => {
    return (
      <fieldset id={id}>
        <label>{label}</label>

        {children}
        {error && value ? (
          <span className={classes.ErrorMessage}>{error}</span>
        ) : null}
      </fieldset>
    );
  };
  const CheckboxComponent = ({
    field: { name, value },
    setFieldValue,
    setTouched,
    datacy,
    label,
    ariaLabel,
    checkboxId,
    touched,
    ariaRequired,
  }) => {
    return (
      <Checkbox
        name={name}
        checked={value}
        datacy={datacy}
        onChange={() => {
          setFieldValue(name, !value);
          setTouched(
            {
              ...touched,
              [name]: true,
            },
            true,
          );
        }}
        checkboxId={checkboxId}
        label={label}
        ariaLabel={ariaLabel}
        ariaRequired={ariaRequired}
      />
    );
  };

  const PhoneNumberComponent = ({
    field: { name, value },
    datacy,
    setFieldValue,
    setTouched,
    touched,
  }) => {
    return (
      <PhoneNumberInput
        required={true}
        name={name}
        current={value}
        datacy={datacy}
        nationalities={props.initData.nationalities}
        onChange={n => {
          setFieldValue(name, n);
          setTouched(
            {
              ...touched,
              [name]: true,
            },
            true,
          );
        }}
      />
    );
  };

  const phoneNumberInputField = (setFieldValue, setTouched, touched) => (
    <div className={classes.InputFieldWrapper}>
      <label>{props.t(`registration.form.phoneNumber`)} *</label>
      <Field
        component={PhoneNumberComponent}
        name={'phoneNumber'}
        value={'phoneNumber'}
        datacy={'input-phoneNumber'}
        setFieldValue={setFieldValue}
        setTouched={setTouched}
        touched={touched}
        type="tel"
        aria-label={props.t(`registration.form.aria.phoneNumber`)}
      />
      <ErrorMessage
        name={'phoneNumber'}
        data-cy={`input-error-phoneNumber`}
        component="span"
        className={classes.ErrorMessage}
      />
    </div>
  );

  const inputField = (name, required, extra, type = 'text', placeholder) => {
    const isEmailInput = name === 'email' || name === 'confirmEmail';
    const handler = isEmailInput ? handleEmailActions : undefined;

    return (
      <FormikInputField
        name={name}
        label={props.t(`registration.form.${name}`)}
        required={required}
        extra={extra}
        type={type}
        autoComplete={isEmailInput ? 'off' : undefined}
        onContextMenu={handler}
        onPaste={handler}
        onCopy={handler}
        onCut={handler}
        placeholder={placeholder || props.t(`registration.form.${name}`)}
      />
    );
  };

  /**
   * Returns default disabled actions for the email inputs
   *
   * @returns false or undefined
   */
  const handleEmailActions = e => {
    e.preventDefault();
    return false;
  };

  const readonlyWhenExistsInput = (name, initialValues, type) =>
    initialValues[name] && initialValues[name].length > 0 ? (
      <div>
        <p className={classes.Label}>{props.t(`registration.form.${name}`)}</p>
        <span>{initialValues[name]}</span>
      </div>
    ) : (
      inputField(name, true, null, type)
    );

  const showExamLang = () => {
    const lang = props.initData.exam_session.language_code;

    return !(lang === 'fin' || lang === 'swe');
  };

  const emptyIfAbsent = value => {
    return value ? value : '';
  };

  const getNationalityDesc = code => {
    const nationality = props.initData.nationalities.find(
      n => n.koodiArvo === code,
    );
    const metadata = nationality.metadata.find(m => m.kieli === 'FI');
    return metadata ? metadata.nimi : '';
  };

  return (
    <Formik
      initialValues={{
        firstName: emptyIfAbsent(props.initData.user.first_name),
        lastName: emptyIfAbsent(props.initData.user.last_name),
        streetAddress: emptyIfAbsent(props.initData.user.street_address),
        zip: emptyIfAbsent(props.initData.user.zip),
        postOffice: emptyIfAbsent(props.initData.user.post_office),
        nationality: props.initData.user.nationalities
          ? props.initData.user.nationalities[0]
          : '',
        ssn: emptyIfAbsent(props.initData.user.ssn),
        birthdate: '',
        gender: '',
        phoneNumber: '358',
        email: emptyIfAbsent(props.initData.user.email),
        confirmEmail: emptyIfAbsent(props.initData.user.email),
        examLang:
          props.initData.exam_session.language_code === 'swe' ? 'sv' : 'fi',
        certificateLang: 'fi',
        personalDataConsent: false,
        termsOfUseConsent: false,
      }}
      validationSchema={validationSchema}
      onSubmit={values => {
        const payload = {
          first_name: values.firstName,
          last_name: values.lastName,
          nationalities: [values.nationality],
          nationality_desc: getNationalityDesc(values.nationality),
          ssn: props.initData.user.ssn || values.ssn,
          birthdate: values.birthdate
            ? moment(values.birthdate, DATE_FORMAT).format(
                ISO_DATE_FORMAT_SHORT,
              )
            : null,
          gender: values.gender,
          certificate_lang: values.certificateLang,
          exam_lang: values.examLang,
          post_office: values.postOffice,
          zip: values.zip,
          street_address: values.streetAddress,
          phone_number: parsePhoneNumberFromString(values.phoneNumber).format(
            'E.164',
          ),
          email: values.email,
          personalDataConsent: values.personalDataConsent,
          termsOfUseConsent: values.termsOfUseConsent,
        };
        props.onSubmitRegistrationForm(props.initData.registration_id, payload);
      }}
      render={({
        values,
        isValid,
        submitCount,
        errors,
        isSubmitting,
        initialValues,
        setFieldValue,
        setTouched,
        touched,
      }) => (
        <Form className={classes.Form} id="form">
          <ScrollToError isValid={isValid} isSubmitting={isSubmitting} />
          <div data-cy="registration-form">
            <p>{props.t('registration.form.info')}</p>
            <div className={classes.InputGroup}>
              {readonlyWhenExistsInput('firstName', initialValues)}
              {readonlyWhenExistsInput('lastName', initialValues)}
            </div>
            <div className={classes.InputGroup}>
              {inputField(
                'streetAddress',
                true,
                null,
                'text',
                props.t('registration.form.streetAddress.placeholder'),
              )}
              <ZipAndPostOffice
                mandatory
                values={values}
                setFieldValue={setFieldValue}
              />
            </div>

            {mobileOrTablet ? (
              <>
                <div className={classes.InputGroup}>
                  {phoneNumberInputField(setFieldValue, setTouched, touched)}
                </div>
                <div className={classes.InputGroup}>
                  {readonlyWhenExistsInput('email', initialValues, 'email')}

                  {!props.initData.user.email && (
                    <>{inputField('confirmEmail', true, null, 'email')}</>
                  )}
                </div>
              </>
            ) : (
              <div className={classes.InputGroup}>
                {phoneNumberInputField(setFieldValue, setTouched, touched)}
                {readonlyWhenExistsInput('email', initialValues, 'email')}
                {!props.initData.user.email && (
                  <>{inputField('confirmEmail', true, null, 'email')}</>
                )}
              </div>
            )}
            {!initialValues.nationality && (
              <div style={{ display: 'flex', flexDirection: 'column' }}>
                <NationalitySelect
                  nationalities={props.initData.nationalities}
                  className={classes.FormSelector}
                />
                <ErrorMessage
                  name={'nationality'}
                  data-cy={`input-error-nationality`}
                  component="span"
                  className={classes.ErrorMessage}
                />
              </div>
            )}
            {!props.initData.user.ssn && (
              <div className={classes.InputGroup}>
                {inputField('birthdate', true)}

                <div style={{ display: 'flex', flexDirection: 'column' }}>
                  <GenderSelect
                    genders={props.initData.genders}
                    className={classes.FormSelector}
                  />
                  <ErrorMessage
                    name={'gender'}
                    data-cy={`input-error-gender`}
                    component="span"
                    className={classes.ErrorMessage}
                  />
                </div>
                <div>
                  {inputField('ssn')}
                  <p> {props.t('registration.form.ssn.text')}</p>
                </div>
              </div>
            )}
            <div className={classes.InputGroup}>
              {showExamLang() && (
                <RadioButtonGroup
                  label={props.t('registration.form.examLang')}
                  value={values.examLang}
                  error={errors.examLang}
                >
                  <div className={classes.RadioButtons}>
                    <Field
                      component={RadioButtonComponent}
                      name="examLang"
                      id={'examLang-fi'}
                      checkedValue={'fi'}
                      label={props.t('common.language.fin')}
                    />
                    <Field
                      component={RadioButtonComponent}
                      name="examLang"
                      id={'examLang-sv'}
                      checkedValue={'sv'}
                      label={props.t('common.language.swe')}
                    />
                  </div>
                </RadioButtonGroup>
              )}

              <RadioButtonGroup
                label={props.t('registration.form.certificateLang')}
                value={values.certificateLang}
                error={errors.certificateLang}
              >
                <div className={classes.RadioButtons}>
                  <Field
                    component={RadioButtonComponent}
                    name="certificateLang"
                    id={'certificateLang-fi'}
                    checkedValue={'fi'}
                    label={props.t('common.language.fin')}
                  />
                  <Field
                    component={RadioButtonComponent}
                    name="certificateLang"
                    id={'certificateLang-sv'}
                    checkedValue={'sv'}
                    label={props.t('common.language.swe')}
                  />
                  <Field
                    component={RadioButtonComponent}
                    name="certificateLang"
                    id={'certificateLang-en'}
                    checkedValue={'en'}
                    label={props.t('common.language.eng')}
                  />
                </div>
              </RadioButtonGroup>
            </div>
          </div>

          <p>
            {props.t('registration.form.specialArrangements.info')}
            <br />
            {props.t('registration.form.specialArrangements.link.info')}:
            <br />
            <ExternalLink
              label={props.t('registration.form.specialArrangements.link.url')}
              url={props.t('registration.form.specialArrangements.link.url')}
            />
          </p>
          <p>{props.t('registration.form.summary.info')}</p>
          <>
            <div className={classes.ConsentContainer}>
              <article>
                <p>
                  <strong>
                    {props.t('registration.form.consent.heading')}
                  </strong>
                </p>
                <p>
                  {props.t('registration.form.consent.info')}
                  <br />
                  <ExternalLink
                    label={props.t('registration.form.consent.ophLink.label')}
                    url={props.t('registration.form.consent.ophLink.url')}
                  />
                </p>
              </article>
              <div className={classes.ConsentCheckbox}>
                <Field
                  component={CheckboxComponent}
                  name={'termsOfUseConsent'}
                  value={'termsOfUseConsent'}
                  checkboxId={'termsOfUseConsent'}
                  datacy={'form-checkbox-terms'}
                  setFieldValue={setFieldValue}
                  touched={touched}
                  setTouched={setTouched}
                  ariaRequired={true}
                  label={props.t('registration.form.consent.confirm')}
                  ariaLabel={props.t('registration.form.consent.confirm')}
                />
                <ErrorMessage
                  name={'termsOfUseConsent'}
                  component="span"
                  className={classes.ErrorMessage}
                />
              </div>
            </div>
            <div className={classes.ConsentContainer}>
              <article>
                <p>
                  <strong>
                    {props.t('registration.form.personalData.consent.heading')}
                  </strong>
                </p>
                <ExternalLink
                  label={props.t('common.yki.consent.description')}
                  url={props.t('common.yki.consent.url')}
                />
              </article>
              <div className={classes.ConsentCheckbox}>
                <Field
                  component={CheckboxComponent}
                  name={'personalDataConsent'}
                  value={'personalDataConsent'}
                  checkboxId={'personalDataConsent'}
                  setTouched={setTouched}
                  touched={touched}
                  setFieldValue={setFieldValue}
                  datacy={'form-checkbox-personal-data'}
                  ariaRequired={true}
                  onClick={() =>
                    setFieldValue(
                      'personalDataConsent',
                      !values.personalDataConsent,
                    )
                  }
                  label={props.t(
                    'registration.form.personalData.consent.confirm',
                  )}
                  ariaLabel={props.t(
                    'registration.form.personalData.consent.confirm',
                  )}
                />
                <ErrorMessage
                  name={'personalDataConsent'}
                  component="span"
                  className={classes.ErrorMessage}
                />
              </div>
            </div>
          </>
          <Button
            type="submit"
            disabled={props.submitting}
            isRegistration={true}
            datacy="form-submit-button"
            ariaLabel={props.t('registration.form.aria.submit.button')}
          >
            {props.t('registration.form.submit.button')}
          </Button>
          {props.submitError && (
            <div data-cy="form-submit-error" className={classes.SubmitError}>
              <RegistrationError
                error={props.submitError}
                defaultKey={'error.registrationForm.submitFailed'}
              />
            </div>
          )}
        </Form>
      )}
    />
  );
};

registrationForm.propTypes = {
  initData: PropTypes.object.isRequired,
  submitting: PropTypes.bool.isRequired,
  onSubmitRegistrationForm: PropTypes.func.isRequired,
  submitError: PropTypes.object,
};

export default withTranslation()(registrationForm);
