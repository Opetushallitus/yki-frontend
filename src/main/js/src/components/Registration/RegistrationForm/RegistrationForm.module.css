.Form input {
  color: hsla(0, 0%, 0%, 1);
}

.RadiobuttonGroup {
  margin: 0px 0 27px 0;
}

.TextInput {
  width: 200px;
  height: 28px;
}

.FormElement {
  margin-top: 20px;
  margin-bottom: 20px;
}

.Form h2 {
  font-size: 1.25em;
  line-height: 28px;
  margin: 8px 0px 14px 0px;
}

.Form h3 {
  margin: 0px 0px 8px 0px;
  line-height: 19px;
}

.Form label {
  font-size: 0.875em;
}

.Form span {
  font-size: 0.875em;
}

.Form input:focus {
  outline: 1px solid hsla(195, 100%, 47%, 1);
}

.Form button {
  display: block;
  width: 252px;
  height: 50px;
  margin: 0 auto;
  border-radius: 2px;
  box-shadow: 0px 0px 4px hsla(0, 0%, 0%, 0.5);
  color: hsl(0, 0%, 100%);
  text-align: center;
  font-size: 0.875em;
  background-color: hsl(96, 96%, 27%);
}

.GenderSelect {
  height: 32px;
  width: 80px;
  border: 1px solid hsl(0, 0%, 68%);
  border-radius: 2px;
  background-color: hsl(0, 0%, 100%);
}

.Birthdate {
  float: left;
  margin-right: 12px;
}

.Birthdate input {
  width: 108px;
  height: 28px;
}

.Birthdate span {
  display: inline;
  margin-left: 4px;
}

.Zip input {
  width: 60px;
  height: 28px;
}

.PostOffice input {
  width: 100px;
  height: 28px;
}

.NationalitySelect {
  height: 32px;
  width: 200px;
  border: 1px solid hsl(0, 0%, 68%);
  border-radius: 2px;
  background-color: hsl(0, 0%, 100%);
}

.ErrorMessage {
  display: block;
  color: hsla(15, 100%, 40%, 1);
  margin: 14px 0px 14px 0px;
}

.SubmitError {
  margin: 14px 0px 14px 0px;
}

@media (min-width: 620px) {
  .Form button {
    margin: 0 40px;
  }
}