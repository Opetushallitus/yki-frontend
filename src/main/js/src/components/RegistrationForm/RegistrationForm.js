import React from 'react';
import PropTypes from 'prop-types';
import * as Yup from 'yup';
import { Formik, Form, Field, ErrorMessage } from 'formik';
import { withNamespaces } from 'react-i18next';

import classes from './RegistrationForm.module.css';
import Button from '../UI/Button/Button';
import RadioButton from '../UI/RadioButton/RadioButton';

const registrationForm = props => {
  const validationSchema = Yup.object().shape({
    firstName: Yup.string().required(props.t('error.mandatory')),
    lastName: Yup.string().required(props.t('error.mandatory')),
    streetAddress: Yup.string().required(props.t('error.mandatory')),
    zip: Yup.string().required(props.t('error.mandatory')),
    postOffice: Yup.string().required(props.t('error.mandatory')),
    phoneNumber: Yup.string().required(props.t('error.mandatory')),
    email: Yup.string().required(props.t('error.mandatory')),
    examLang: Yup.string().required(props.t('error.mandatory')),
    certificateLang: Yup.string().required(props.t('error.mandatory')),
  });

  const RadioButtonComponent = ({
    field: { name, value, onChange },
    id,
    checkedValue,
    label,
  }) => {
    return (
      <RadioButton
        name={name}
        id={id}
        checkedValue={checkedValue}
        value={value}
        onChange={onChange}
        label={label}
      />
    );
  };

  const RadioButtonGroup = ({
    value,
    id,
    label,
    className,
    children,
    error,
  }) => {
    return (
      <div className={className} id={id}>
        <h3>{label}</h3>
        {children}
        {error && value ? (
          <span className={classes.ErrorMessage}>{error}</span>
        ) : null}
      </div>
    );
  };

  const readonlyField = (name, initialValues) => (
    <React.Fragment>
      <h3>{props.t(`registration.form.${name}`)}</h3>
        <span>{initialValues[name]}</span>
    </React.Fragment>
  );

  const inputField = name => (
    <React.Fragment>
      <h3>{props.t(`registration.form.${name}`)}</h3>
      <Field
        name={name}
        data-cy={`input-${name}`}
        className={classes.TextInput}
      />
      <ErrorMessage
        name={name}
        component="span"
        className={classes.ErrorMessage}
      />
    </React.Fragment>
  );

  const readonlyWhenExistsInput = (name, initialValues) => (
    <React.Fragment>
      <h3>{props.t(`registration.form.${name}`)}</h3>
      {initialValues[name] && initialValues[name].length > 0 ? (
        <span>{initialValues[name]}</span>
      ) : (
        inputField(name)
      )}
    </React.Fragment>
  );

  return (
    <Formik
      initialValues={{
        firstName: props.formInitData.user.first_name,
        lastName: props.formInitData.user.last_name,
        streetAddress: props.formInitData.user.street_address,
        zip: props.formInitData.user.zip,
        postOffice: props.formInitData.user.post_office,
        phoneNumber: '',
        email: '',
        examLang: 'fi',
        certificateLang: 'fi',
      }}
      validationSchema={validationSchema}
      onSubmit={values => {
        console.log(values);
      }}
      render={({ values, isValid, errors, initialValues }) => (
        <Form className={classes.Form}>
          <div data-cy="registration-form">
            <p>{props.t('registration.form.info')}</p>
            <div className={classes.FormElement}>
              {readonlyWhenExistsInput('firstName', initialValues)}
            </div>
            <div className={classes.FormElement}>
              {readonlyWhenExistsInput('lastName', initialValues)}
            </div>
            <div className={classes.FormElement}>
              {readonlyField('streetAddress', initialValues)}
            </div>
            <div className={classes.FormElement}>
              {readonlyField('zip', initialValues)}{' '}
              <span>{initialValues['postOffice']}</span>
            </div>
            <div className={classes.FormElement}>
              {inputField('phoneNumber', initialValues)}
            </div>
            <div className={classes.FormElement}>
              {inputField('email', initialValues)}
            </div>
            <div
              className={[classes.FormElement, classes.RadiobuttonGroup].join(
                ' ',
              )}
            >
              <RadioButtonGroup
                label={props.t('registration.form.examLang')}
                value={values.examLang}
                error={errors.examLang}
              >
                <Field
                  component={RadioButtonComponent}
                  name="examLang"
                  id={'examLang-fi'}
                  checkedValue={'fi'}
                  label={props.t('common.language.fin')}
                />
                <Field
                  component={RadioButtonComponent}
                  name="examLang"
                  id={'examLang-sv'}
                  checkedValue={'sv'}
                  label={props.t('common.language.swe')}
                />
              </RadioButtonGroup>
            </div>
            <div
              className={[classes.FormElement, classes.RadiobuttonGroup].join(
                ' ',
              )}
            >
              <RadioButtonGroup
                label={props.t('registration.form.certificateLang')}
                value={values.certificateLang}
                error={errors.certificateLang}
              >
                <Field
                  component={RadioButtonComponent}
                  name="certificateLang"
                  id={'certificateLang-fi'}
                  checkedValue={'fi'}
                  label={props.t('common.language.fin')}
                />
                <Field
                  component={RadioButtonComponent}
                  name="certificateLang"
                  id={'certificateLang-sv'}
                  checkedValue={'sv'}
                  label={props.t('common.language.swe')}
                />
                <Field
                  component={RadioButtonComponent}
                  name="certificateLang"
                  id={'certificateLang-en'}
                  checkedValue={'en'}
                  label={props.t('common.language.eng')}
                />
              </RadioButtonGroup>
            </div>
          </div>
          <Button type="submit" disabled={false}>
            Ilmoittaudu
          </Button>
        </Form>
      )}
    />
  );
};

registrationForm.propTypes = {
  formInitData: PropTypes.object.isRequired,
  // onSubmit: PropTypes.func.isRequired,
};

export default withNamespaces()(registrationForm);
