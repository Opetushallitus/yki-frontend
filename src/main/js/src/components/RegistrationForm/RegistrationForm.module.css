.RadiobuttonGroup {
  margin: 0px 0 27px 0;
}

.MaxParticipantsInput {
  display: block;
  width: 35px;
  height: 35px;
  padding-left: 0px;
  text-align: center;
}

.TextInput {
  width: 200px;
  height: 28px;
}

.Select {
  height: 35px;
  width: 250px;
  border: 1px solid hsl(0, 0%, 68%);
  border-radius: 2px;
  background-color: hsl(0, 0%, 100%);
}

.FormElement {
  margin-top: 20px;
  margin-bottom: 20px;
}

.Form h2 {
  font-size: 1.25em;
  line-height: 28px;
  margin: 8px 0px 14px 0px;
}

.Form h3 {
  margin: 0px 0px 8px 0px;
  line-height: 19px;
}

.Form label {
  font-size: 0.875em;
}

.Form span {
  font-size: 0.875em;
}

.Form input:focus {
  outline: 1px solid hsla(195, 100%, 47%, 1);
}

.Form button {
  display: block;
  width: 252px;
  height: 50px;
  margin: 0 auto;
  border-radius: 2px;
  box-shadow: 0px 0px 4px hsla(0, 0%, 0%, 0.5);
  color: hsl(0, 0%, 100%);
  text-align: center;
  font-size: 0.875em;
  background-color: hsl(96, 96%, 27%);
}

.ErrorMessage {
  display: block;
  color: hsla(15, 100%, 40%, 1);
  margin: 14px 0px 14px 0px;
}

.SubmitError {
  margin: 14px 0px 14px 0px;
}

@media (min-width: 620px) {
  .StartingText {
    margin-bottom: 79px;
  }

  .Form button {
    margin: 0 114px;
  }
}