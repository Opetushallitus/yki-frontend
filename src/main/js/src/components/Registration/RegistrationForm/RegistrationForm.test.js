import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import { Provider } from 'react-redux';
import { createStore, combineReducers } from 'redux';
import userReducer from '../../../store/reducers/user';

import { registrationForm as RegistrationForm } from './RegistrationForm';

configure({ adapter: new Adapter() });

jest.mock('react-i18next', () => ({
  withTranslation: () => Component => {
    Component.defaultProps = { ...Component.defaultProps, t: () => '' };
    return Component;
  },
}));

const initData = {
  user: {
    last_name: 'Parkkonen-Testi',
    nick_name: null,
    ssn: '260553-959D',
    post_office: 'HELSINKI',
    street_address: 'Atel√§niitynpolku 29 G',
    first_name: 'Carl-Erik',
    zip: '00100',
    nationalities: ['246'],
  },
  exam_session: {
    language_code: 'eng',
  },
};
describe('<RegistrationForm />', () => {
  const mockStore = createStore(combineReducers({ user: userReducer }));
  it('should render registration form', () => {
    const wrapper = shallow(
      <Provider store={mockStore}>
        <RegistrationForm
          t={key => key}
          i18n={{ lang: 'fi' }}
          initData={initData}
          submitting={false}
          onSubmitRegistrationForm={jest.fn()}
        />
      </Provider>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
