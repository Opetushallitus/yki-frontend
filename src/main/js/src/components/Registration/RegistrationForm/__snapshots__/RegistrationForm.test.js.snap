// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RegistrationForm /> should render registration form 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Object {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <registrationForm
    i18n={
      Object {
        "lang": "fi",
      }
    }
    initData={
      Object {
        "exam_session": Object {
          "language_code": "eng",
        },
        "user": Object {
          "first_name": "Carl-Erik",
          "last_name": "Parkkonen-Testi",
          "nationalities": Array [
            "246",
          ],
          "nick_name": null,
          "post_office": "HELSINKI",
          "ssn": "260553-959D",
          "street_address": "AtelÃ¤niitynpolku 29 G",
          "zip": "00100",
        },
      }
    }
    onSubmitRegistrationForm={[MockFunction]}
    submitting={false}
    t={[Function]}
  />
</ContextProvider>
`;
